connector/dialect.tdd

```diff
@@ -379,650 +183,568 @@
+      <formula>CAST(%1 AS LONG)</formula>
+      <argument type='int' />
     </function>
-    <function group='aggregate' name='COUNTD' return-type='int'>
-      <formula>COUNT(DISTINCT %1)</formula>
-      <unagg-formula>(CASE WHEN %1 IS NULL THEN 0 ELSE 1 END)</unagg-formula>
+    <function group='cast' name='LONG' return-type='int'>
+      <formula>CAST(%1 AS LONG)</formula>
       <argument type='bool' />
     </function>
-    <function group='aggregate' name='COUNTD' return-type='int'>
-      <formula>COUNT(DISTINCT %1)</formula>
-      <unagg-formula>(CASE WHEN %1 IS NULL THEN 0 ELSE 1 END)</unagg-formula>
-      <argument type='real' />
-    </function>
-    <function group='aggregate' name='COUNTD' return-type='int'>
-      <formula>COUNT(DISTINCT %1)</formula>
-      <unagg-formula>(CASE WHEN %1 IS NULL THEN 0 ELSE 1 END)</unagg-formula>
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='str' />
     </function>
-    <function group='aggregate' name='COUNTD' return-type='int'>
-      <formula>COUNT(DISTINCT %1)</formula>
-      <unagg-formula>(case when %1 IS NULL THEN 0 ELSE 1 END)</unagg-formula>
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='datetime' />
     </function>
-    <function group='aggregate' name='COVAR' return-type='real'>
-      <formula>COVAR_SAMP(%1, %2)</formula>
-      <unagg-formula>NULL</unagg-formula>
-      <argument type='real' />
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='real' />
     </function>
-    <function group='aggregate' name='COVARP' return-type='real'>
-      <formula>COVAR_POP(%1, %2)</formula>
-      <unagg-formula>(CASE WHEN %1 IS NULL THEN NULL WHEN %2 IS NULL THEN NULL ELSE 0.0 END)</unagg-formula>
-      <argument type='real' />
-      <argument type='real' />
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
+      <argument type='int' />
     </function>
-    <function group='aggregate' name='MAX' return-type='bool'>
-      <formula>BOOL_OR(%1)</formula>
-      <unagg-formula>%1</unagg-formula>
+    <function group='cast' name='STR' return-type='str'>
+      <formula>CAST(%1 AS STRING)</formula>
       <argument type='bool' />
     </function>
-    <function group='aggregate' name='MAX' return-type='real'>
-      <formula>MAX(%1)</formula>
-      <unagg-formula>%1</unagg-formula>
-      <argument type='real' />
+    <function group='date' name='CURRENT_TIMESTAMP' return-type='datetime'>
+      <formula>CURRENT_TIMESTAMP</formula>
     </function>
-    <function group='aggregate' name='MAX' return-type='int'>
-      <formula>MAX(%1)</formula>
-      <unagg-formula>%1</unagg-formula>
+    <function group='date' name='CURRENT_TIMESTAMP' return-type='datetime'>
+      <formula>CURRENT_TIMESTAMP %1</formula>
```

Comment:
If I understand this correctly, this is from specifying the precision, right?
From the syntax doc, it should be CURRENT_TIMESTAMP(%1). If CURRENT_TIMESTAMP %1 is also a valid MongoSQL syntax, then you can keep it as is of course (I personally prefer to see the parenthesis because it means less possibly error prone generated queries).

Reply:
Good catch, I'll add the parenthesis back for CURRENT_TIMESTAMP.  It must have been part of my pass of removing parenthesis.